src/shared_modules/utils/tests/reflectiveJson_test.cpp

```diff
@@ -0,0 +1,218 @@
+    bool validPermutation2 = (result == R"({"key2":"value2","key1":"value1"})");
+    EXPECT_TRUE(validPermutation1 || validPermutation2);
+}
+
+TEST_F(ReflectiveJsonTest, NumericValues)
+{
+    std::unordered_map<std::string, int> map {{"intVal", 42}, {"zeroVal", 0}, {"negativeVal", -10}};
+    std::string result = jsonFieldToString(map);
+
+    // Possible orders in the JSON:
+    // {"intVal":42,"zeroVal":0,"negativeVal":-10}
+    // {"intVal":42,"negativeVal":-10,"zeroVal":0}
+    // etc.
+    // We check presence of all fields with correct values.
+    EXPECT_NE(result.find(R"("intVal":42)"), std::string::npos);
+    EXPECT_NE(result.find(R"("zeroVal":0)"), std::string::npos);
+    EXPECT_NE(result.find(R"("negativeVal":-10)"), std::string::npos);
+    EXPECT_EQ(result.front(), '{');
+    EXPECT_EQ(result.back(), '}');
+}
+
+TEST_F(ReflectiveJsonTest, NestedMap)
+{
+    std::unordered_map<std::string, std::unordered_map<std::string, int>> map {{"outerKey", {{"innerKey", 123}}}};
+    std::string result = jsonFieldToString(map);
+
+    EXPECT_EQ(result, "{\"outerKey\":{\"innerKey\":123}}");
+}
+
+TEST_F(ReflectiveJsonTest, EscapedStringValue)
+{
+    std::unordered_map<std::string, std::string> map {{"key", "He said \"Hello\""}};
+
+    std::string result = jsonFieldToString(map);
+
+    EXPECT_NE(result.find(R"("key":"He said \"Hello\"")"), std::string::npos);
+}
+
+TEST_F(ReflectiveJsonTest, BasicStructSerialization)
+{
+    TestData<float> obj;
+    obj.fieldOne = "001";
+    obj.fieldTwo = 30;
+    obj.fieldThree = 1.3;
+
+    std::string json;
+    serializeToJSON(obj, json);
+
+    EXPECT_NE(json.find(R"("fieldOne":"001")"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldTwo":30)"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldThree":1.3)"), std::string::npos);
+}
+
+TEST_F(ReflectiveJsonTest, NestedStructSerialization)
+{
+    struct NestedData
+    {
+        std::string innerField;
+        int innerValue;
+
+        REFLECTABLE(MAKE_FIELD("innerField", &NestedData::innerField),
+                    MAKE_FIELD("innerValue", &NestedData::innerValue));
+    };
+
+    TestData<NestedData> obj;
+    obj.fieldOne = "Outer";
+    obj.fieldTwo = 2;
+    obj.fieldThree.innerField = "Inner";
+    obj.fieldThree.innerValue = 42;
+
+    std::string json;
+    serializeToJSON(obj, json);
+
+    EXPECT_NE(json.find(R"("fieldOne":"Outer")"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldThree":{"innerField":"Inner","innerValue":42})"), std::string::npos);
+}
+
+TEST_F(ReflectiveJsonTest, BasicStructSerializationWithEmptyFields)
+{
+    TestData<std::string_view> obj;
+    obj.fieldOne = "";
+    obj.fieldTwo = 0;
+    obj.fieldThree = "";
+
+    std::string json;
+    serializeToJSON(obj, json);
+
+    EXPECT_EQ(json.find(R"("fieldOne":)"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldTwo":0)"), std::string::npos);
+    EXPECT_EQ(json.find(R"("fieldThree":)"), std::string::npos);
+}
```

Comment:
Could we check the empty value here also?

Reply:
We tested using `EXPECT_EQ`, and according to the current implementation behavior, the value is not set if it is empty. However, we can discuss this during today's daily meeting and determine if a refactor is necessary.

Reply:
Currently this is the output

```cpp
{"fieldTwo":0}
```

If the value is empty, is not added to JSON

Reply:
I confirmed that this is the expected behavior, due indexer doesn't allow empty values :)

Reply:
> I confirmed that this is the expected behavior, due indexer doesn't allow empty values :)

This is interesting.
But @Dwordcito , should this JSON util behave like this?

Reply:
This utility is intended to have the highest possible performance, writing empty spaces implies writing keys and quotes, saving us from those two operations is an improvement. For our use case we don't mind the printing of voids yet, there could be a variant in the future when the need arises.